examples/token-classification/run_tf_ner.py

```diff
@@ -203,8 +203,8 @@ def compute_metrics(p: EvalPrediction) -> Dict:
     trainer = TFTrainer(
         model=model,
         args=training_args,
-        train_dataset=train_dataset.get_dataset() if train_dataset else None,
-        eval_dataset=eval_dataset.get_dataset() if eval_dataset else None,
+        train_sized_dataset=(train_dataset.get_dataset(), len(train_dataset)) if train_dataset else (None, None),
```

Comment:
No, don't need to know the batch size or whatever else, just the number of examples. Anyway, this is done just because I got some errors on running the current way to compute the size on TPUs, because it cannot compute Python functions such as lambda.

This is also not definitive.

Reply:
Did the original solution to compute the dataset size works on your side with TPUs?

Reply:
Yes, the original solution (`self.train_dataset.reduce(tf.constant(0), lambda x, _: x + 1).numpy()`) worked on my side. I'm using TF2.2, with TPU on GCP.

Reply:
Ok, good to know then, it works for TF >= 2.2 only.

Reply:
That's weird. I tried with ctpu with TF 2.2 and I get the following error:

```
2020-06-19 12:33:47.347085: W tensorflow/core/distributed_runtime/eager/remote_tensor_handle_data.cc:76] Unable to destroy remote tensor handles. If you are running a tf.function, it usually indicates some op in the graph gets an error: No registered 'PyFunc' OpKernel for 'CPU' devices compatible with node {{node PyFunc}}
	.  Registered:  <no registered kernels>

	 [[PyFunc]]
Traceback (most recent call last):
  File "run_tf_ner.py", line 281, in <module>
    main()
  File "run_tf_ner.py", line 213, in main
    trainer.train()
  File "/home/jplu/transformers/src/transformers/trainer_tf.py", line 276, in train
    train_ds = self.get_train_tfdataset()
  File "/home/jplu/transformers/src/transformers/trainer_tf.py", line 73, in get_train_tfdataset
    self.num_train_examples = self.train_dataset.reduce(tf.constant(0), lambda x, _: x + 1).numpy()
  File "/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/ops.py", line 961, in numpy
    maybe_arr = self._numpy()  # pylint: disable=protected-access
  File "/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/ops.py", line 929, in _numpy
    six.raise_from(core._status_to_exception(e.code, e.message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.NotFoundError: No registered 'PyFunc' OpKernel for 'CPU' devices compatible with node {{node PyFunc}}
	.  Registered:  <no registered kernels>

	 [[PyFunc]]
```